jib-cli/src/main/java/com/google/cloud/tools/jib/cli/CommonContainerConfigCliOptions.java

```diff
@@ -0,0 +1,179 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.cli;
+
+import com.google.cloud.tools.jib.api.Ports;
+import com.google.cloud.tools.jib.api.buildplan.AbsoluteUnixPath;
+import com.google.cloud.tools.jib.api.buildplan.ImageFormat;
+import com.google.cloud.tools.jib.api.buildplan.Port;
+import com.google.common.collect.ImmutableSet;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import picocli.CommandLine;
+
+/** Common command line options shared between jar and war command. */
+public class CommonContainerConfigCliOptions {
+
+  @CommandLine.Option(
+      names = "--from",
+      paramLabel = "<base-image>",
+      description = "The base image to use.")
+  @SuppressWarnings("NullAway.Init") // initialized by picocli
+  private String from;
+
+  @CommandLine.Option(
+      names = "--expose",
+      paramLabel = "<port>",
+      split = ",",
+      description = "Ports to expose on container, example: --expose=5000,7/udp.")
+  private List<String> exposedPorts = Collections.emptyList();
+
+  @CommandLine.Option(
+      names = "--volumes",
+      paramLabel = "<volume>",
+      split = ",",
+      description =
+          "Directories on container to hold extra volumes,  example: --volumes=/var/log,/var/log2.")
+  private List<String> volumes = Collections.emptyList();
+
+  @CommandLine.Option(
+      names = "--environment-variables",
+      paramLabel = "<key>=<value>",
+      split = ",",
+      description =
+          "Environment variables to write into container, example: --environment-variables env1=env_value1,env2=env_value2.")
+  private Map<String, String> environment = Collections.emptyMap();
+
+  @CommandLine.Option(
+      names = "--labels",
+      paramLabel = "<key>=<value>",
+      split = ",",
+      description =
+          "Labels to write into container metadata, example: --labels=label1=value1,label2=value2.")
+  private Map<String, String> labels = Collections.emptyMap();
+
+  @CommandLine.Option(
+      names = {"-u", "--user"},
+      paramLabel = "<user>",
+      description = "The user to run the container as, example: --user=myuser:mygroup.")
+  @SuppressWarnings("NullAway.Init") // initialized by picocli
+  private String user;
+
+  @CommandLine.Option(
+      names = {"--image-format"},
+      defaultValue = "Docker",
+      paramLabel = "<image-format>",
+      description =
+          "Format of container, candidates: ${COMPLETION-CANDIDATES}, default: ${DEFAULT-VALUE}.")
+  @SuppressWarnings("NullAway.Init") // initialized by picocli
+  private ImageFormat format;
+
+  @CommandLine.Option(
+      names = "--program-args",
```

Comment:
Maybe this shouldn't be included, which has a similar nature to `--jvm-args` in that it's a factor in computing a final entrypoint?

Reply:
Oh good catch.

Reply:
Ah, I forgot that `programArguments` corresponds to `CMD` in `Dockerfile`, which is different from `ENTRYPOINT`.